### PR TITLE
Refresh markdown preview on all file changes

### LINKED ISSUE #138304: Markdown preview not auto-updating after modifying SVGs in RemoteHub workspaces
From @rebornix:

> I was using vscode.dev for writing notebook arch docs
> which references an svg file nex to the md. the svg can be edited via drawio extension but due to the lack of file watcher on vscode.dev, the markdown preview won't be updated after the svg is modified
> this is probably the only noticeable difference between desktop and vscode.dev for documentation editing

### PR DESCRIPTION
**Pull request by joyceerhl**
This PR fixes https://github.com/microsoft/vscode/issues/138304

### CODE REVIEW

**Review by hediet — APPROVED** [review]
Looks good, but I don't know if some places assume that uri to be a file uri.

**Review by raghavthind2005 — APPROVED** [review]

**Comment by joyceerhl** [comment]
@hediet are there specific uri schemes that we want to ignore? E.g @JacksonKearl mentioned we probably don't want to watch https uris for changes.

Will also wait for @mjbvz's feedback on this.

**Comment by hediet** [comment]
> are there specific uri schemes that we want to ignore?

Probably, but an exclusion lists always feels a little bit fishy, as it is very easy to miss something.

File: extensions/markdown-language-features/src/features/preview.ts [hunk 1]
```diff
@@ -427,7 +427,7 @@ class MarkdownPreview extends Disposable implements WebviewResourceProvider {
 		const root = vscode.Uri.joinPath(this._resource, '../');
 		for (const src of srcs) {
 			const uri = urlToUri(src, root);
-			if (uri && uri.scheme === 'file' && !this._fileWatchersBySrc.has(src)) {
+			if (uri && !this._fileWatchersBySrc.has(src)) {
```
Comment (mjbvz) [inline extensions/markdown-language-features/src/features/preview.ts:430]: Yes I'd say as a start let's ignore `http`, `https`, and `data`. We can add more if needed

**Comment by joyceerhl** [comment]
Added https://github.com/microsoft/vscode/pull/138301/commits/e5810c8173e5c86f7969efa48f010f1b20ee2003 to ensure we don't watch image src uris with https, http or data schemes.

image

**Comment by mjbvz** [comment]
Thanks!

**Pull request merged**